interp/syntax_def.ml

```diff
@@ -105,3 +105,10 @@ let search_syntactic_definition ?loc kn =
   let def = out_pat pat in
   verbose_compat ?loc kn def v;
   def
+
+let search_filtered_syntactic_definition ?loc filter kn =
+  let pat,v = KNmap.find kn !syntax_table in
+  let def = out_pat pat in
+  let res = filter def in
+  (match res with Some _ -> verbose_compat ?loc kn def v | None -> ());
+  res
```

Comment:
Code duplication not quite pretty. Why not export the warning function instead and let the client decide what to do with the result?

Reply:
This was indeed the alternative but I thought it would be more intrusive. Do you prefer I change?

Reply:
My own arbitration was not to factorize, so, considering that a change would take time, I'd do it (only) if you push stronglier than what you're doing in the direction of factorizing.

Reply:
Fine with me. But you should still fix the test as explained in the above review comment though.

Reply:
It is fixed I think: I'm now using a local notation.